### PR TITLE
Display default value of boolean properly

### PR DESCRIPTION
**Pull request by tajo**
- fixes https://github.com/cloudflare/doca/issues/22
- immutable added to deps so this lib can be tested/developed through `npm link`
- will have to open the same PR in doca-cf-theme

### CODE REVIEW

File: components/constraints.js [hunk 1]
```diff
@@ -12,8 +12,11 @@ class Constraints extends Component {
     if (!constraints) return <div />;
     return (
       <ul style={{ listStyle: 'none', paddingLeft: 0 }}>
-        {(constraints.get('default') || constraints.get('default') === 0) &&
-          <li>default value: {constraints.get('default')}</li>
+        {constraints.has('default') &&
+          (constraints.get('type') === 'boolean' ?
+            (constraints.get('default') ? 'default value: true' : 'default value: false') :
```
Comment (handrews) [inline components/constraints.js]: I'm not 100% sure I'm reading this right- maybe it's the cough medicine but I'm finding the nested ?: to be a little challenging.

constraints.get('type') checks the value of the schema's "type" property, right?  Or is it checking  a JavaScript runtime type.  Assuming it's the schema field, what happens if type is set to ["boolean", "null"] or is unset?  A boolean default is still allowed in those cases.

What happens if we just do

{constraints.has('default') && <li>default value: {constraints.get('default')}

If we want to ensure that defaults are valid against their schemas, we should do that elsewhere.

Comment (tajo) [reply components/constraints.js]: - First it checks if schema's "default" property exists.
- If it does, it checks if the schema's "type" property is boolean.
  - If it is boolean, it checks if the value of schema's "default" property is true.
    - if yes, it displays: `default value: true`
    - if no, it displays: `default value: false`
  - If it isn't boolean, it just displays the value as `default value: constraints.get('default')`

**Examples:**

```json
"in_stock": {
  "type": "boolean",
  "description": "Whether the product is in stock or not",
  "example": true,
  "default": false
},
```

displays:

```
default value: false
```

```json
"in_stock": {
  "type": "boolean",
  "description": "Whether the product is in stock or not",
  "example": true,
  "default": true
},
```

displays:

```
default value: true
```

```json
"in_stock": {
  "type": "boolean",
  "description": "Whether the product is in stock or not",
  "example": true
},
```

displays:

```
// nothing
```

```json
"in_stock": {
  "type": "number",
  "description": "Whether the product is in stock or not",
  "example": true,
  "default": 0
},
```

displays:

```
default value: 0
```

Comment (tajo) [reply components/constraints.js]:
```
default value: {constraints.get('default')}
```

We had partly this before and that was the reason of this bug. Because JavaScript/Immutable.js doesn't translate `false` into string and prints nothing.

Comment (handrews) [reply components/constraints.js]: Ah, so false prints nothing.  OK, so the important part though is you should not be testing for `"type": "boolean"` in the schema, but as the default value itself.  There is no requirement that a schema with `true` or `false` as a default value has its `"type"` field set to `"boolean"`.

Comment (tajo) [reply components/constraints.js]: Ok, so you can have a schema with property "type" missing? If not, it doesn't make sense to have default values for `true` of `false` for other types than `boolean`?

Comment (handrews) [reply components/constraints.js]: All schema properties are optional.  The empty schema `{}` validates everything.  So for instance, this:

```JSON
{
    "default": false
}
```

Is a schema that accepts any value of any type, with no restriction whatsoever.  But it's default value is boolean `false`.

That's pretty straightforward, I think- sometimes you actually do have unrestricted data, if only rarely.

A weirder but still valid schema is:

```JSON
{
    "type": "integer",
    "default": false
}
```

In this example, the default does not validate.  Which is weird, and of questionable value.  It gets discussed a lot over on the spec project- `"default"` is a bit of a headache because it's not very well-specified as to how you can and can't use it.  See json-schema-org/json-schema-spec#204 and json-schema-org/json-schema-spec#217 for way more discussion than you probably want to know.

A more realistic schema that you would not handle with the current code is 

```JSON
{
    "type": ["boolean", "null"],
    "default": false
}
```

which declares a 3-state field (nullable boolean).  
Basically, a boolean default value should be presented the same way whether the type is present or not, regardless of the type specified.  It's a bit weird, but that's what it is right now.

**Comment by tajo** [comment]
@handrews Check it out again. My previous solution was silly. Now, it's much simpler and it doesn't rely on type property at all. (I'll update doca-cf-theme when this is merged).

**Review by handrews — APPROVED** [review]
